redash/query_runner/query_results.py

```diff
@@ -60,21 +65,29 @@ def _load_query(user, query_id):
     return query
 
 
-def create_tables_from_query_ids(user, connection, query_ids):
-    for query_id in set(query_ids):
+def get_query_results(user, query_id, bring_from_cache):
         query = _load_query(user, query_id)
+        if bring_from_cache and query.latest_query_data_id is not None:
```

Comment:
When there are no results stored and the user asked for cache we shouldn't fallback to running the query but rather show an error message saying this ("No cached result available for query xxx.").

Reply:
Also seems like the indentation here is a bit off.

Reply:
Sounds right, i fixed it